outlines/blaze.md

```diff
@@ -0,0 +1,56 @@
+# Blaze
+
+1. Introduction to Blaze -- the tracker-backed handlebars like templating syntax
+  1. Example of spacebars syntax, data context + helpers
+  2. Data contexts and access - name lookup (`.`, `this`, `..`, `[0]`), and null values
+  3. Helpers, arguments, options
+  4. Inclusion of other templates + arguments, passing data contexts
+  5. Helpers inside tags -- returning strings and objects, `checked={{isChecked}}`
+  6. Nested helpers + sub expressions
+  7. Block helpers (you can create them with templates, see 10. below)
+  8. Safestrings and `{{{`
+  9. Builtin block helpers
+    1. `{{#if/unless}}`
+    2. `{{#with}}`
```

Comment:
Oh, now I remembered why I was unable to implement this previously: https://github.com/meteor/meteor/issues/4494

Because my simple implementation would be something like:

```javascript
Template.registerHelper('lookup', function (name) {
  var computedField = new ComputedField(function () {
    var value = Template.instance().view.lookup(name);
    if (_.isFunction(value)) {
      value = value();
    }
    return value;
  }
  return computedField();
});
```

But, autorun does not have access to template instance. :-( In this particular case this could be solved with:

```javascript
Template.registerHelper('lookup', function (name) {
  const view = Template.instance().view;
  var computedField = new ComputedField(function () {
    var value = view.lookup(name);
    if (_.isFunction(value)) {
      value = value();
    }
    return value;
  }
  return computedField();
});
```

The downside of this approach of course is that template helpers are always bound to data context, so this would rerun every time data context changes. Maybe to transition away from current Blaze, a new type of helpers could be defined? Like "methods"? Which would be the same as helpers, just that they have `this` set to the template instance and no automatic data context binding?